src/main/scala/sigmastate/lang/SigmaTyper.scala

```diff
@@ -345,6 +345,7 @@ class SigmaTyper(val builder: SigmaBuilder) {
     case v: EvaluatedValue[_] => v
     case v: SigmaBoolean => v
     case v: Upcast[_, _] => v
+    case v @ Select(_, _, Some(_)) => v
```

Comment:
@greenhat looks suspicious

Reply:
I made this shortcut for typed Select cause for the expression "1L * 1.toLong" `bimap` runs `r` through typer but it's already typed in [MethodCall handling](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/db66cb4065aa37ac4b3763266e135b10b5256631/src/main/scala/sigmastate/lang/SigmaTyper.scala#L178)